packages/espressocash_app/lib/features/outgoing_link_payments/services/recover_pending_watcher.dart

```diff
@@ -0,0 +1,134 @@
+import 'dart:async';
+
+import 'package:collection/collection.dart';
+import 'package:get_it/get_it.dart';
+import 'package:injectable/injectable.dart';
+import 'package:solana/dto.dart';
+import 'package:solana/solana.dart';
+import 'package:uuid/uuid.dart';
+
+import '../../../../config.dart';
+import '../../../../core/amount.dart';
+import '../../../../core/currency.dart';
+import '../../../../core/escrow_private_key.dart';
+import '../../activities/data/transaction_repository.dart';
+import '../../tokens/token.dart';
+import '../data/repository.dart';
+import '../models/outgoing_link_payment.dart';
+
+@injectable
+class RecoverPendingWatcher implements Disposable {
```

Comment:
I would say, this service should run just once on app start / after restoring wallet, doesn't make sense to listen to updates afterwards.